storage/options.go

```diff
@@ -280,7 +281,8 @@ func (o *options) SetEncodingM3TSZPooled() Options {
 		SetDatabaseBlockOptions(opts.blockOpts).
 		SetContextPool(contextPool).
 		SetEncoderPool(encoderPool).
-		SetMultiReaderIteratorPool(multiReaderIteratorPool)
+		SetMultiReaderIteratorPool(multiReaderIteratorPool).
+		SetIdentifierPool(opts.IdentifierPool())
```

Comment:
Hm, maybe we should just call `NewSeriesOptionsFromOptions` here?  e.g. `seriesOpts := NewSeriesOptionsFromOptions(opts)`?